tests/test_normalized_optimizer.py

```diff
@@ -0,0 +1,266 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from absl.testing import absltest, parameterized
+
+from emerging_optimizers.riemannian_optimizers.normalized_optimizer import ObliqueAdam, ObliqueSGD
+
+
+class NormalizedOptimizerFunctionalTest(parameterized.TestCase):
+    """Tests for ObliqueSGD and ObliqueAdam optimizers that preserve row/column norms."""
+
+    def setUp(self):
+        """Set random seed before each test."""
+        # Set seed for PyTorch
+        torch.manual_seed(1234)
+        # Set seed for CUDA if available
+        if torch.cuda.is_available():
+            torch.cuda.manual_seed_all(1234)
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Don't auto select device, especially in test. If the test is accidently assigned to a machine without GPU, it can end up never being tested on GPU. Same for other test cases.